client/client.go

```diff
@@ -0,0 +1,604 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package client
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/rpc"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+	"text/tabwriter"
+	"time"
+
+	"github.com/ciao-project/ciao/osprepare"
+	"github.com/intel/ccloudvm/types"
+	"github.com/pkg/errors"
+)
+
+type logger struct{}
+
+func (l logger) V(int32) bool {
+	return false
+}
+
+func (l logger) Infof(s string, args ...interface{}) {
+	out := fmt.Sprintf(s, args...)
+	fmt.Print(out)
+	if !strings.HasSuffix(out, "\n") {
+		fmt.Println()
+	}
+}
+
+func (l logger) Warningf(s string, args ...interface{}) {
+	l.Infof(s, args)
+}
+
+func (l logger) Errorf(s string, args ...interface{}) {
+	l.Infof(s, args)
+}
+
+const systemdService = `
+[Unit]
+Description=Configurable CloudVM Service
+
+[Service]
+Type=simple
+ExecStart=%s/bin/ccvm
+KillMode=process
+`
+
+const systemdSocket = `
+[Socket]
+ListenStream=%s/.ccloudvm/socket
+
+[Install]
+WantedBy=sockets.target
+`
+
+func getGoPath() (string, error) {
+	goPathBytes, err := exec.Command("go", "env", "GOPATH").Output()
+	if err != nil {
+		return "", errors.Wrap(err, "Unable to determine GOPATH")
+	}
+	return strings.TrimSpace(string(goPathBytes)), nil
+}
+
+// Setup Installs dependencies
+func Setup(ctx context.Context) error {
+	home := os.Getenv("HOME")
+	if home == "" {
+		return errors.New("HOME is not defined")
+	}
+	home = strings.TrimSpace(home)
+
+	goPath, err := getGoPath()
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("Installing host dependencies")
+	osprepare.InstallDeps(ctx, ccloudvmDeps, logger{})
+
+	systemdRootPath := filepath.Join(home, ".local/share/systemd/user")
+	err = os.MkdirAll(systemdRootPath, 0700)
+	if err != nil {
+		return errors.Wrap(err, "Unable to create systemd directory")
+	}
+
+	servicePath := filepath.Join(systemdRootPath, "ccloudvm.service")
+	serviceData := fmt.Sprintf(systemdService, goPath)
+	err = ioutil.WriteFile(servicePath, []byte(serviceData), 0600)
+	if err != nil {
+		return errors.Wrap(err, "Unable to write service file")
+	}
+
+	socketPath := filepath.Join(systemdRootPath, "ccloudvm.socket")
+	socketData := fmt.Sprintf(systemdSocket, home)
+	err = ioutil.WriteFile(socketPath, []byte(socketData), 0600)
+	if err != nil {
+		return errors.Wrap(err, "Unable to write service file")
+	}
+
+	err = exec.Command("systemctl", "--user", "daemon-reload").Run()
```

Comment:
for security reasons, I think that you should specify the full path of systemctl

Reply:
I'm reluctant to do this as the command would then only work if systemctl is installed in the location we hardcode on every machine that ccloudvm is installed on.  Also, I don't see what the threat is here.  Someone would have to have local access to the shell on which the end user called ccloudvm setup to change his PATH.